docs/configuration.md

```diff
@@ -1600,6 +1600,34 @@ Apart from these, the following properties are also available, and may be useful
     How many batches the Spark Streaming UI and status APIs remember before garbage collecting.
   </td>
 </tr>
+<tr>
+  <td><code>spark.streaming.driver.writeAheadLog.closeFileAfterWrite</code></td>
+  <td>false</td>
+  <td>
+    Whether to close the file after writing a write ahead log record on the driver. Because S3 doesn't
+    support flushing of data, when using S3 for checkpointing, you should enable it to achieve read
+    after write consistency.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.streaming.receiver.writeAheadLog.closeFileAfterWrite</code></td>
+  <td>false</td>
+  <td>
+    Whether to close the file after writing a write ahead log record on the receivers. Because S3
+    doesn't support flushing of data, when using S3 for checkpointing, you should enable it to
+    achieve read after write consistency.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.streaming.driver.writeAheadLog.allowBatching</code></td>
+  <td>true</td>
+  <td>
+    Whether to batch write ahead logs on the driver to write. When using S3 for checkpointing, write
+    operations on the driver usually take too long. Enabling batching write ahead logs will improve
```

Comment:
Whether to batch writes to the metadata WAL in the driver. This is useful from improving performance for file systems like S3 where the write latency is high and / or scenarios with large number of receivers.